src/components/generic/TreeView.tsx

```diff
@@ -85,129 +86,140 @@ class TreeView extends React.Component<Props, State> {
+        parent.taxonomy_academy.length === 0
+      ) {
+        treeData = [
+          {
+            key: `${ current.id }`,
+            label: `${ current.title ? current.title.rendered : "" }`,
+            link: `${ current.link }`,
+            academyPage: current.taxonomy_academy && current.taxonomy_academy.length > 0 ? true : false,
+            nodes: treeData
           }
+        ];
+        break;
+      }
+      layer.reverse();
+      treeData = layer.map(node => {
+        if (node.key === `${ current.id }`) {
+          return {
+            ...node,
+            nodes: treeData
+          };
         }
-        if (menu.child_items) {
-          this.findAcademyPage(menu.child_items);
-        }
+        return node;
       });
-    }
-  }
-
-  /**
-   * Creates array from menu structure
-   *
-   * @param menus menu structure
-   */
-  private menuStructureToArray = (menus: MenuItemData[]): MenuItemData[] => {
-    const array: MenuItemData[] = [];
-    menus.forEach(menu => {
-      array.push(menu);
-      if (menu.child_items) {
-        const child_array = this.menuStructureToArray(menu.child_items);
-        child_array.forEach(child => {
-          array.push(child);
-        });
+      if (current.parent === 0 || !parent) {
+        break;
       }
+      current = parent;
+    }
+    this.setState({
+      treeData: treeData,
+      initialOpenNodes: initialOpenNodes
     });
-    return array;
+    this.fetchTreeChildren();
   }
 
   /**
-   * Builds link tree structure
-   *
-   * @param menuItems menu item data array
+   * Fetches child nodes for entire tree
    */
-  private buildTree = (menuItems: MenuItemData[], opened: string[]): any => {
-    const { schoolBlogPages, page } = this.state;
-    const linkTreeStructure: LinkTreeStructure[] = [];
-    menuItems.forEach(item => {
-      if ((schoolBlogPages && !schoolBlogPages.find(blog => blog.id === item.object_id)) || !schoolBlogPages) {
-        linkTreeStructure.push(
-          {
-            key: `${ item.object_id }`,
-            label: item.title || "",
-            link: item.url || "",
-            nodes: this.showChildren(item) ? this.buildTree(item.child_items ? item.child_items : [], [...opened, `${ opened.length > 0 ? opened[opened.length - 1] + "/" : ""}${ item.object_id }`]) : undefined
-          }
-        );
-      }
-      if (page && `${ page.id }` === item.object_id) {
-        this.setState({
-          initialOpenNodes: opened
-        });
+  private fetchTreeChildren = async () => {
+    const api = ApiUtils.getApi();
+    const { treeData, onAcademyPage } = this.state;
+    const fetchChildren = async (parent: LinkTreeStructure): Promise<LinkTreeStructure[]> => {
+      const children = await api.getWpV2Pages({ parent: [`${ parent.key }`], per_page: 100 });
+      if (!children || (!onAcademyPage && parent.academyPage)) {
+        return [];
       }
+      const promises = children.map( async node => {
```

Comment:
No need for async since map doesn't contain any asynchonous calls
```suggestion
      const promises = children.map(node => {
```